crates/egui-winit/src/lib.rs

```diff
@@ -939,21 +939,49 @@ pub fn update_viewport_info(
         }
     };
 
-    viewport_info.focused = Some(window.has_focus());
-    viewport_info.fullscreen = Some(window.fullscreen().is_some());
-    viewport_info.inner_rect = inner_rect;
-    viewport_info.monitor_size = monitor_size;
-    viewport_info.native_pixels_per_point = Some(window.scale_factor() as f32);
-    viewport_info.outer_rect = outer_rect;
-    viewport_info.title = Some(window.title());
+    update_info.title = Some(window.title());
+    update_info.native_pixels_per_point = Some(window.scale_factor() as f32);
 
+    update_info.monitor_size = monitor_size;
+    update_info.inner_rect = inner_rect;
+    update_info.outer_rect = outer_rect;
+
+    /*
+    // Deprecated
     if cfg!(target_os = "windows") {
         // It's tempting to do this, but it leads to a deadlock on Mac when running
         // `cargo run -p custom_window_frame`.
         // See https://github.com/emilk/egui/issues/3494
-        viewport_info.maximized = Some(window.is_maximized());
         viewport_info.minimized = Some(window.is_minimized().unwrap_or(false));
+        viewport_info.maximized = Some(window.is_maximized());
     }
+    */
+
+    update_info.minimized = match cfg!(target_os = "windows") {
+        true => Some(window.is_minimized().unwrap_or(false)),
+        false => window.is_minimized(),
```

Comment:
As the comment above says, this leads to deadlock on Mac

Reply:
> As the comment above says, this leads to deadlock on Mac

Okay, I was curious about the results.